ts/lib/socket.ts

```diff
@@ -76,9 +78,22 @@ export class ConstellationSocket extends EventEmitter {
 
         this.once('event:hello', () => {
             this.ready = true;
+            if (this.reconnecting) {
+                this.reconnecting = false;
+                this.emit("reopen");
+            }
             this.queue.forEach(data => this.send(data));
             this.queue = [];
         });
+        this.on("close", () => {
+            if (!this.options.autoReconnect) {
+                return;
+            }
+            this.reconnecting = true;
+            setTimeout(() => {
```

Comment:
So you'd rather a method for closing a socket gracefully which won't trigger the reconnect; via setting a boolean / status enum? Otherwise it would trigger a reconnect.